src/vim9expr.c

```diff
@@ -811,7 +873,34 @@ compile_load(
 	    }
 	}
 	if (gen_load)
-	    res = generate_LOAD(cctx, ISN_LOAD, idx, NULL, type);
+	{
+	    // Optimization: check if can generate LOAD CLASSMEMBER.
+	    int did_classmember = FALSE;
+	    if (type->tt_type == VAR_OBJECT
+		&& (type->tt_class->class_flags & CLASS_INTERFACE) == 0
+		&& !gen_load_outer)
+	    {
+		// Got an object that is not an interface.
+		int is_class;
+		char_u *name_end;
+		ocmember_T *m;
+		int idx01;
+		if (find_class_object_member(type->tt_class, end_arg,
+					    &name_end, &idx01, &is_class, &m)
+		    && is_class
+		    && m->ocm_access != VIM_ACCESS_PRIVATE)
```

Comment:
For readability reasons, it will be good to separate out the call to the "find_class_object_member()" function to a separate if statement followed by the check for "is_class" and "m->ocm_access".